src/shared/containers/ResourceViewContainer.tsx

```diff
@@ -67,14 +67,46 @@ function constructUnDeprecateUrl(
   latestResource: Resource,
   orgLabel: string,
   projectLabel: string
-) {
-  return `${apiEndpoint}/${
-    resource!['@type'] === 'File' ? 'files' : 'resources'
-  }/${orgLabel}/${projectLabel}/${
-    resource!['@type'] === 'File' ? '' : '_/'
-  }${encodeURIComponent(resource!['@id'])}/undeprecate?rev=${
-    latestResource!._rev
-  }`;
+): string {
+  const typePathMapping: { [key: string]: string } = {
+    File: 'files',
+    Storage: 'storages',
+    ElasticSearchView: 'views',
+    SparqlView: 'views',
+    CompositeView: 'views',
+    View: 'views',
+    Schema: 'schemas',
+  };
+
+  const determineResourcePathSegment = (
+    inputTypes: string | string[]
+  ): string => {
+    const types = Array.isArray(inputTypes) ? inputTypes : [inputTypes];
+
+    for (const type of types) {
+      if (type in typePathMapping) {
+        return typePathMapping[type];
+      }
+    }
+
+    return 'resources';
+  };
+
+  const primaryResourceType = resource['@type'] || '';
+  const resourcePathSegment = determineResourcePathSegment(primaryResourceType);
+  const slashPrefix = Array.isArray(primaryResourceType)
+    ? primaryResourceType.some(type =>
+        Object.keys(typePathMapping).includes(type)
+      )
+      ? ''
+      : '_/'
+    : Object.keys(typePathMapping).includes(primaryResourceType)
```

Comment:
Nice to have: To keep it consistent with line 87 above should we use `primaryResourceType in typePathMapping` here? It is also slightly better in terms of memory consumption than creating a new array with `Object.keys`